pcsx2-qt/Debugger/MemoryViewWidget.cpp

```diff
@@ -277,9 +277,11 @@ bool MemoryViewTable::KeyPress(int key, QChar keychar)
 
 		if (keyCharIsText)
 		{
-			InsertIntoSelectedHexView(((u8)QString(QChar(key)).toInt(&pressHandled, 16)));
+			// Check if key pressed is hex before insertion (QString conversion fails otherwise)
+			u8 keyPressed = (u8)QString(QChar(key)).toInt(&pressHandled, 16);
 			if (pressHandled)
 			{
+				InsertIntoSelectedHexView(keyPressed);
```

Comment:
Hmm should the assigning (and declaration I guess) of keyPressed not be inside the if? like, what is pressHandled is null? The conversion will fail/throw an error potentially.  Also there's not much point in having it outside the if, if it's not used elsewhere.

You could also make it const.

Reply:
> Hmm should the assigning (and declaration I guess) of keyPressed not be inside the if? like, what is pressHandled is null? The conversion will fail/throw an error potentially if it's not checked for assignment first. Also there's not much point in having it outside the if, if it's not used elsewhere.
> 
> You could also make it const.

If you're referring to the function call being inside of the 'if' block, that's because this would make that block dead code (it will never evaluate to true, as pressHandled is the status bit of our QString conversion). If you mean putting it in the condition itself, we could say ```if (u8 keyPressed = (u8)QString(QChar(key)).toInt(&pressHandled, 16) >= 0 && pressHandled)```, and this should work since we'll evaluate the conditional left to right and thus the assignment completes by the time 'pressHandled' is evaluated, but it feels far less readable.

That being said, I could make it const; after all, it won't be being used again. Would you like me to go ahead and do that, ref?

Reply:
oh that function is assigning 16 to pressHandled? my bad.

Reply:
The function that converts the key from QString toInt assigns the boolean reference passed in to a true or false depending on if the conversion worked (It won't if it's a non-hex string character). So it's setting the boolean the following if statement uses.

Personally I think it'd be less confusing to create a new boolean with a name that clarifies that it's validating the success of the conversion like `ok`, `isOK`, `isValidHex` but is probably fine.

Reply:
> oh that function is assigning 16 to pressHandled? my bad.

My understanding of the passed &pressHandled is that it acts as a status bit for the conversion, while 16 acts as the numerical base. QString::toInt documentation can be seen [here](https://doc.qt.io/qt-6/qstring.html#toInt). Previously what we were doing is writing to the hex viewer by default no matter if the conversion succeeded or failed (it just returns 0 on failure) and then entering the conditional statement depending on the success/failure of the QString conversion.

Reply:
> The function that converts the key from QString toInt assigns the boolean reference passed in to a true or false depending on if the conversion worked (It won't if it's a non-hex string character). So it's setting the boolean the following if statement uses.
> 
> Personally I think it'd be less confusing to create a new boolean with a name that clarifies that it's validating the success of the conversion like `ok`, `isOK`, `isValidHex` but is probably fine.

It needs to set press handled because that communicates to the caller that this function handled the press.

Reply:
I see, good call. Makes sense with that context.